tests/conformance/state/state.go

```diff
@@ -453,6 +453,100 @@ func ConformanceTests(t *testing.T, props map[string]string, statestore state.St
 				}
 			}
 		})
+
+		t.Run("transaction-order", func(t *testing.T) {
+			// Arrange
+			firstKey := "key1"
+			firstValue := "value1"
+			secondKey := "key2"
+			secondValue := "value2"
+			thirdKey := "key3"
+			thirdValue := "value3"
+
+			// for CosmosDB
+			partitionMetadata := map[string]string{
+				"partitionKey": "myPartition",
+			}
+
+			// prerequisite: key1 should be present
+			err := statestore.Set(&state.SetRequest{
+				Key:      firstKey,
+				Value:    firstValue,
+				Metadata: partitionMetadata,
+			})
+			assert.NoError(t, err, "set request should be successful")
+
+			// prerequisite: key2 should not be present
+			err = statestore.Delete(&state.DeleteRequest{
+				Key:      secondKey,
+				Metadata: partitionMetadata,
+			})
+			assert.NoError(t, err, "delete request should be successful")
+
+			// prerequisite: key3 should not be present
+			err = statestore.Delete(&state.DeleteRequest{
+				Key:      thirdKey,
+				Metadata: partitionMetadata,
+			})
+			assert.NoError(t, err, "delete request should be successful")
+
+			operations := []state.TransactionalStateOperation{
+				// delete an item that already exists
+				{
+					Operation: state.Delete,
+					Request: state.DeleteRequest{
+						Key: firstKey,
+					},
+				},
+				// upsert a new item
+				{
+					Operation: state.Upsert,
+					Request: state.SetRequest{
+						Key:   secondKey,
+						Value: secondValue,
+					},
+				},
+				// delete the item that was just upserted
+				{
+					Operation: state.Delete,
+					Request: state.DeleteRequest{
+						Key: secondKey,
+					},
+				},
+				// upsert a new item
+				{
+					Operation: state.Upsert,
+					Request: state.SetRequest{
+						Key:   thirdKey,
+						Value: thirdValue,
+					},
+				},
+			}
+
+			expected := map[string][]byte{
```

Comment:
nit : How about reword this from `expected` to `expectedKeys` or `expectedKeysMap`, any thoughts?

Reply:
would suggest `expectedValues`